### ISSUE TITLE
🐞 Env Variable as empty string blows up

### ISSUE THREAD

**Issue by jpadams** [dagger/dagger#4821]
### What is the issue?

🐛 If you have an env var that is an empty string, Graphql explodes as it's converted to null or undefined. To fix it you have to add whitespace

![image](https://user-images.githubusercontent.com/3187222/227260732-e74a2922-9847-4e69-8fa3-c2cc1fc56f45.png)

### Log output

```console
Field "withEnvVariable" argument "value" of type "String!" is required but not provided.
```

### Steps to reproduce

```javascript
import * as dagger from '@dagger.io/dagger';

dagger.connect(async (client) => {
  const out = await client
    .container()
    .from('alpine')
    .withEnvVariable('FOO', '')
    .withExec(['sh', '-c', 'env | grep FOO'])
    .stdout();
  console.log(out)
})
```

This produces the following query, which is missing the value:

<pre>
{
    container  { 
        from (address: "alpine") { 
            withEnvVariable (<b>name: "FOO"</b>) {
                withExec (args: ["sh","-c","env | grep FOO"]) { 
                    stdout
                }
            }
        }
    }
}
</pre>

### SDK version

NodeJS v0.4.2

**Comment by helderco** [comment]
It seems there is a problem in NodeJS and Python SDKs. 

## NodeJS

If value is empty, it's not being included in the request:

<pre>
{
    container  { 
        from (address: "alpine") { 
            withEnvVariable (<b>name: "FOO"</b>) {
                withExec (args: ["sh","-c","env | grep FOO"]) { 
                    stdout
                }
            }
        }
    }
}
</pre>

```shell
Field "withEnvVariable" argument "value" of type "String!" is required but not provided.
```

<details>
<summary>Full code</summary>

```javascript
import * as dagger from '@dagger.io/dagger';

dagger.connect(async (client) => {
  const out = await client
    .container()
    .from('alpine')
    .withEnvVariable('FOO', '')
    .withExec(['sh', '-c', 'env | grep FOO'])
    .stdout();
  console.log(out)
})
```
</details>

## Python

The serialization is somehow converting the empty string to an empty list:

<pre>
╭───────────────────────────────────── <class 'collections.deque'> ─────────────────────────────────────╮
│ deque([iterable[, maxlen]]) --> deque object                                                          │
│                                                                                                       │
│ ╭───────────────────────────────────────────────────────────────────────────────────────────────────╮ │
│ │ deque([                                                                                           │ │
│ │ │   Field(type_name='Query', name='container', args={}),                                          │ │
│ │ │   Field(type_name='Container', name='from', args={'address': 'alpine'}),                        │ │
│ │ │   Field(type_name='Container', name='withEnvVariable', args={'name': 'FOO', <b>'value': []</b>}),      │ │
│ │ │   Field(                                                                                        │ │
│ │ │   │   type_name='Container',                                                                    │ │
│ │ │   │   name='withExec',                                                                          │ │
│ │ │   │   args={'args': ['sh', '-c', 'env | grep FOO']}                                             │ │
│ │ │   ),                                                                                            │ │
│ │ │   Field(type_name='Container', name='stdout', args={})                                          │ │
│ │ ])                                                                                                │ │
│ ╰───────────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                       │
│ maxlen = None                                                                                         │
╰───────────────────────────────────────────────────────────────────────────────────────────────────────╯
</pre>

```console
graphql.error.graphql_error.GraphQLError: String cannot represent value: []
```


<details>
<summary>Full code</summary>

```python
import anyio
import dagger


async def main():
    async with dagger.Connection() as client:
        out = await (
            client.container()
            .from_("alpine")
            .with_env_variable("FOO", "")
            .with_exec(["sh", "-c", "env | grep FOO"])
            .stdout()
        )

    print(out)


anyio.run(main)
```
</details>

**Comment by helderco** [comment]
Fixed Python in #4884, now this is focused on NodeJS.
